src/bootstrap/doc.rs

```diff
@@ -481,22 +482,27 @@ impl Step for Std {
         let my_out = builder.crate_doc_out(target);
         t!(symlink_dir_force(&builder.config, &my_out, &out_dir));
 
-        let mut cargo = builder.cargo(compiler, Mode::Std, target, "doc");
-        compile::std_cargo(builder, &compiler, target, &mut cargo);
+        let run_cargo_rustdoc_for = |package: &str| {
+            let mut cargo = builder.cargo(compiler, Mode::Std, target, "rustdoc");
```

Comment:
You'll need to add `rustdoc` wherever we do `== "doc"` inside `fn cargo` (in builder.rs).

I think that might be enough, not sure; if not I can investigate more.

Reply:
I'll make the changes and then we'll see. Thanks!